This needs a LOT more work, particularly in the build system

--- origsrc/webkitgtk-2.0.4/Source/WebKit2/UIProcess/API/gtk/WebKitWebContext.cpp	2013-07-19 02:04:19.000000000 -0500
+++ src/webkitgtk-2.0.4/Source/WebKit2/UIProcess/API/gtk/WebKitWebContext.cpp	2014-07-11 01:39:24.925604600 -0500
@@ -184,7 +184,7 @@ static CString injectedBundleDirectory()
 
 static CString injectedBundleFilename()
 {
-    GOwnPtr<char> bundleFilename(g_build_filename(injectedBundleDirectory().data(), "libwebkit2gtkinjectedbundle.so", NULL));
+    GOwnPtr<char> bundleFilename(g_module_build_path(injectedBundleDirectory().data(), "webkit2gtkinjectedbundle"));
     return bundleFilename.get();
 }
 
--- origsrc/webkitgtk-2.0.4/Source/WebKit2/UIProcess/Launcher/gtk/ProcessLauncherGtk.cpp	2013-07-19 02:04:19.000000000 -0500
+++ src/webkitgtk-2.0.4/Source/WebKit2/UIProcess/Launcher/gtk/ProcessLauncherGtk.cpp	2014-07-11 01:31:28.492105300 -0500
@@ -43,6 +43,8 @@
 
 #if OS(LINUX)
 #include <sys/prctl.h>
+#endif
+#if OS(UNIX)
 #include <sys/socket.h>
 #endif
 
